h2o-py/h2o/estimators/estimator_base.py

```diff
@@ -108,24 +108,30 @@ def train(self, x=None, y=None, training_frame=None, offset_column=None, fold_co
         :param bool verbose: Print scoring history to stdout. Defaults to False.
         """
         self._train(x=x, y=y, training_frame=training_frame, offset_column=offset_column, fold_column=fold_column,
-                    weights_column=weights_column, validation_frame=validation_frame, max_runtime_secs=max_runtime_secs, 
-                    ignored_columns=ignored_columns, model_id=model_id, verbose=verbose)
+                    weights_column=weights_column, validation_frame=validation_frame, max_runtime_secs=max_runtime_secs,
+                    ignored_columns=ignored_columns, model_id=model_id, verbose=verbose, **kwargs)
 
 
     def _train(self, x=None, y=None, training_frame=None, offset_column=None, fold_column=None,
-              weights_column=None, validation_frame=None, max_runtime_secs=None, ignored_columns=None,
-              model_id=None, verbose=False, extend_parms_fn=None):
+               weights_column=None, validation_frame=None, max_runtime_secs=None, ignored_columns=None,
+               model_id=None, verbose=False, extend_parms_fn=None, **kwargs):
         has_default_training_frame = hasattr(self, 'training_frame') and self.training_frame is not None
         training_frame = H2OFrame._validate(training_frame, 'training_frame',
                                             required=self._requires_training_frame() and not has_default_training_frame)
         validation_frame = H2OFrame._validate(validation_frame, 'validation_frame')
         algo = self.algo
         parms = self._parms.copy()
         if(algo == "targetencoder"):
-            x = parms["encoded_columns"]
-            y = parms["target_column"]
+            if ("data_leakage_handling" in kwargs):
```

Comment:
I think it's debatable whether `data_leakage_handling` belongs in the constructor or the train method... but I think we just put arguments that relate to the given `training_frame` in `.train()`.  So my instinct is to keep `data_leakage_handling` in the constructor since it's a characteristic of the TE process, rather than the frame you are operating on.

Reply:
Discussed over Slack yesterday, removed from train method's arguments.